apps/oracles/eth-rpc/src/lib.rs

```diff
@@ -144,128 +144,198 @@ type MyProvider = alloy::providers::fillers::FillProvider<
+                    sum += x;
+                    weight += 1.0_f64;
+                }
+                Err(e) => error = Some(DataFeedResultValue::Error(e.to_string())),
+            };
+        }
+        let value = if sum > 0.0_f64 {
+            let div = if let Some(div) = resourses
+                .iter()
+                .find(|x| x.feed_id == *feed_id)
+                .and_then(|x| x.arguments.divisor)
+            {
+                weight * div
+            } else {
+                weight
+            };
+            DataFeedResultValue::Numerical(sum / div)
+        } else {
+            match error {
+                Some(err) => err,
+                None => DataFeedResultValue::Error("No data".to_string()),
+            }
         };
-        payload.values.push(DataFeedResult {
+        let data_feed_result = DataFeedResult {
             id: feed_id.to_string(),
             value,
-        });
+        };
+        payload.values.push(data_feed_result);
     }
     Ok(payload)
 }
 
-fn print_results(results: &FetchedDataForFeed, payload: &Payload) {
-    let mut keys = results.keys().cloned().collect::<Vec<u64>>();
-    keys.sort();
+fn print_responses(results: &FetchedDataHashMap) {
+    let mut feed_ids = results.keys().cloned().collect::<Vec<u64>>();
+    feed_ids.sort();
 
     let mut table = Table::new();
     table.set_format(*format::consts::FORMAT_NO_LINESEP_WITH_TITLE);
 
     table.set_titles(Row::new(vec![
         Cell::new("Feed ID").style_spec("bc"),
+        Cell::new("Label").style_spec("bc"),
         Cell::new("Method").style_spec("bc"),
-        Cell::new("Value").style_spec("bc"),
+        Cell::new("Raw response").style_spec("bc"),
         Cell::new("RPC Url").style_spec("bc"),
+        Cell::new("Contract").style_spec("bc"),
     ]));
 
-    for key in keys {
-        let description = match key {
-            100001 => "YnETHx -> convertToAssets",
-            100002 => "YToken -> exchangeRate",
-            100003 => "YnBNBx -> convertToAssets",
-            _ => "unknown",
-        };
-        let rpc_url = format!("{:?}", results.get(&key).unwrap().rpc_url);
-        let value = payload
+    for feed_id in feed_ids {
+        let res = results.get(&feed_id).unwrap();
+        for (resp, contact) in zip(&res.responses, &res.contacts) {
+            let rpc_url = format!("{:?}", resp.rpc_url.clone());
+            let value = format!("{:?}", resp.result_as_f64());
+            table.add_row(Row::new(vec![
+                Cell::new(&feed_id.to_string()).style_spec("r"),
+                Cell::new(&contact.label).style_spec("l"),
+                Cell::new(&contact.method_name).style_spec("r"),
+                Cell::new(&value).style_spec("r"),
+                Cell::new(&rpc_url).style_spec("r"),
+                Cell::new(&contact.address.to_string()).style_spec("r"),
+            ]));
+        }
+    }
+
+    table.printstd();
+}
+
+fn print_payload(payload: &Payload, resourses: &[FeedConfig]) {
```

Comment:
It would be nice to have the option to turn off pretty printing and only leave necessary logs when deployed. I guess we are using this pretty printing for debugging purposes?